diff --git a/node_modules/react-native-fast-tflite/src/expo-plugin/withCoreMLDelegate.ts b/node_modules/react-native-fast-tflite/src/expo-plugin/withCoreMLDelegate.ts
index 18648b0..69ca4d8 100644
--- a/node_modules/react-native-fast-tflite/src/expo-plugin/withCoreMLDelegate.ts
+++ b/node_modules/react-native-fast-tflite/src/expo-plugin/withCoreMLDelegate.ts
@@ -12,7 +12,7 @@ import { insertContentsAtOffset } from '@expo/config-plugins/build/utils/commonC
 
 export const withCoreMLDelegate: ConfigPlugin = (config) =>
   withPlugins(config, [
-    // Add $EnableCoreMLDelegate = true to the top of the Podfile
+    // Add $EnableCoreMLDelegate = true to the top of the Podfile (only if not already present)
     [
       withDangerousMod,
       [
@@ -23,14 +23,26 @@ export const withCoreMLDelegate: ConfigPlugin = (config) =>
             'Podfile'
           )
 
-          const prevPodfileContent = fs.readFileSync(podfilePath, 'utf-8')
-          const newPodfileContent = insertContentsAtOffset(
-            prevPodfileContent,
-            '$EnableCoreMLDelegate=true\n',
-            0
-          )
+          const lineToAdd = '$EnableCoreMLDelegate=true'
+          const podfileContent = fs.readFileSync(podfilePath, 'utf-8')
+
+          // Vérifie si la ligne existe déjà
+          const alreadyExists = podfileContent
+            .split('\n')
+            .some((line) => line.trim() === lineToAdd)
+
+          if (!alreadyExists) {
+            const newPodfileContent = insertContentsAtOffset(
+              podfileContent,
+              `${lineToAdd}\n`,
+              0
+            )
 
-          fs.writeFileSync(podfilePath, newPodfileContent)
+            fs.writeFileSync(podfilePath, newPodfileContent)
+            console.log(`✅ Ajout de "${lineToAdd}" au Podfile`)
+          } else {
+            console.log(`ℹ️ "${lineToAdd}" déjà présent dans le Podfile, pas de modification.`)
+          }
 
           return currentConfig
         },
